### DISCUSSION TITLE
Square brackets in primary key from postgress ext source

### DISCUSSION THREAD

**Discussion by andhie** [Budibase/budibase#2629] [Help] [answered]
To test out Bidibase, i used a Postgres with PgAdmin docker for Budibase to use.
I have a `Patient` table with the `_id` with data type serial (it will auto create sequence), non-null, primary key via the pgadmin portal.

Below is the screenshot of the data queried out on PgAdmin
![Screenshot 2021-09-17 at 12 02 16 PM](https://user-images.githubusercontent.com/3997169/133723067-1eb42283-0a68-4164-9fb3-359c692190c1.jpg)

Below is the screenshot of Bidibase of the patient table. Note that it includes `%5B` and `%5D` which are `[` and `]` respectively.
![Screenshot 2021-09-17 at 12 02 40 PM](https://user-images.githubusercontent.com/3997169/133723070-1b93814f-78e9-43e3-87b5-3cbafa466bfd.jpg)

Is this expected? as it causes issue for me when using `{{ URL.id }}` and trying to query another data.
i.e. `/patient/:id/vitals` but in the url it becomes `/patient/%5B1%5D/vitals`, in turn will not match any Patient ID in my Vitals table

**Comment by mjashanks** [comment, marked as answer]
Hi @andhie 

The column name `_id` is a special name in Budibase. I'm not actually sure how we should handle a scenario where your database has a column called `_id`, but i'd consider this a bug in Budibase as it is a valid scenario.

If possible, could you rename your identity column? I'll use  `your_new_id`, so I'm super clear! 

Then, in Budibase, you should change the "View" link on your table  to point to `/patient/{{ patient Table.patient.your_new_id }}/vitals`. Yours might differ slightly, but you are using the `your_new_id` column, rather than the `_id`.

This should result in the URL `/patient/1/vitals`

**Reply by andhie** [reply]
thanks for the response. i have changed the the `_id` to `id` instead in the postgres. it now displaying proper integer value.

I did create the `/patient/1/vitals` screen, now the url is correct. 

1. For this screen, i accidentally enabled "Create link in navigation bar", i dont see any options to remove it from the nav bar. is this function available?

2. I've also written a custom Query to query out patient specific vitals data (due to internal db doesnt support filter on relationship column). When i run the website, it show a red toast "An error has occured" and in the network inspector, the real error was `{"message":"error: syntax error at or near \";\"","status":500}`. Is there a way for me to test/check the binding results? or even know what was the query that was send to postgres?
![Screenshot 2021-09-17 at 8 01 07 PM](https://user-images.githubusercontent.com/3997169/133779466-0f10bd5e-ddc2-47e1-8ab8-371bc551637e.jpg)

**Reply by mjashanks** [reply]
Sure..

1. Every screen has a "Layout". The layout is where the Nav bar lives. https://docs.budibase.com/building-apps/design/layouts 
2. The default parameter is expecting a value. change `{{ URL.id }}` to `1`

Unfortunately, there is no way to debug this properly at the moment, but I think that's your issue.

**Reply by andhie** [reply]
For the second issue, do you mean its not possible to query the data based on dynamic criteria? like in this case where i try to get the `{{ URL.id }}` data to be used for SQL filtering. 

I use the binding in the Create Query part in the data where its hinted in  https://docs.budibase.com/building-apps/design/binding, it seems to support SQL query binding. (added 1 as default value as you mentioned, but still having errors)
![Screenshot 2021-09-17 at 8 51 56 PM](https://user-images.githubusercontent.com/3997169/133785592-8bad3e3c-f30e-496c-8048-45cdf8a056ee.jpg)

**Reply by mjashanks** [reply]
sorry  - you can. But that's not where you put `{{ URL.id }}`

In the Design section, on your Data Provider, you will select your custom query. Then, click the little settings button. This will popup a drawer that allows you to input your parameter value:

![image](https://user-images.githubusercontent.com/3524181/133787329-5bc8a170-0a38-49a3-8154-407f8798cd45.png)

**Reply by andhie** [reply]
thanks a lot, this screenshot is really helpful.
the changes i need to make was 
1. i had `:id` as my parameter and in the sql i used `id` without the colon
2. the sql should not have semicolon to end it

would it be ok if i enhance the documentation to better show how to use custom query with sample screenshot especially.

**Comment by mjashanks** [comment]
Yes - you are welcome to, thank you!
